installation/docker.md

```diff
@@ -0,0 +1,94 @@
+
+The container by defaults runs all the LangStream components, a Kafka Broker and a S3 service (using Minio).
+
+The docker container exposes the Control Plane on the default port (8090) and the API Gateway on the default port (8091),
+this way you can run most of the CLI commands against the local container, especially the commands to interact with the gateway.
+
+When you kill the application with Ctrl-C the environment is automatically disposed.
+If you need to persist your topics or the S3 environment then you have to build your own instance.yaml file and pass it using the "-i" flag.
+
+### Selecting the services to run
+
+By default the docker container runs all the LangStream components, a Kafka Broker and a S3 service (using Minio).
+
+You can use the following flags to select the services to run:
+
+* --start-broker true|false: starts the Kafka broker
+* --start-s3 true|false: starts the S3 service
+* --start-webservices true|false: starts the LangStream HTTP components (control plane and API gateway)
+
+For instance if you are using an external Apache Kafka or Pulsar broker you don't need to start Kafka in the container and you can save local resources.
+
+### Running a single agent
+
+By default the "docker run" mode runs all the agents in the application.
+If you want to debug or work on a single agent you can use the "--only-agent" flag to specify the agent to run.
+
+The id of the agent is the same id of the "executors" section in the application descriptor that you can see in the "langstram apps get -o yaml" command.
+Basically it is not the id of an agent in the pipeline.yaml file but it is the id of a pyshical executor. When you run your application in a Kubernetes cluster
+the id would be the id of a Statefulset.
+
+### Execution model
+
+When you run your application in "docker run" mode, the container runs a simplified enviroment that doesn't need Kubernetes, but this comes with a fews simplifications to the execution runtime.
+
+In production mode, on Kubernetes, the LangStream planner builds and execution plan from your pipeline files, and then it submits the execution plan to the Kubernetes cluster in the form of Statefulsets. Each Statefulset is responsible to run a single agent and you can usually configure the number of replicas for each agent in the pipeline file, in the "resources" section.
+
+But in docker mode there is only one Java process that runs all the agents, and for each agent it starts only one execution flow, like having a statefulset with only one replica.
+The resources (JVM and CPU) are shared between all the agents, so if you have a lot of agents in your application you may need to increase the resources of the docker container.
+
+The initialisation of the assets is always performed, for all the assets declared in the application, independently from the agents that are running. This is because the assets are shared between all the agents, even if they are declared in some pipeline file.
```

Comment:
```suggestion
The initialisation of the assets is always performed independently from the agents that are running, for all the assets declared in the application. This is because the assets are shared between all the agents, even if they are declared in some pipeline file.
```